### ISSUE TITLE
ansible playbook over tor with reboots

### ISSUE THREAD

**Issue by dolanjs** [freedomofpress/securedrop#1008]
Experiencing intermittent timeout errors after the ansible playbook reboots the servers.

1) The playbook is being ran over tor
2) A reboot is required during the anisble playbook
3) The [task](https://github.com/freedomofpress/securedrop/blob/develop/install_files/ansible-base/roles/common/tasks/apply_grsec_lock.yml#L27) to wait for the ssh port to become available again completes successfully
4) The [task after](https://github.com/freedomofpress/securedrop/blob/develop/install_files/ansible-base/roles/common/tasks/apply_grsec_lock.yml#L48) waiting for the ssh port to be available timeouts
- This issue only applies when the ansible playbook is being ran over tor so does not affect initial installs where the playbook is not being ran over tor.
- The issue only applies if the conditions requiring a reboot (new grsec kernel or other security updates that require a reboot) are met
- This issue only happens intermittently. To test you could create the reboot-required file on the app/mon servers that the playbook looks for to require a reboot.

**Comment by dolanjs** [comment]
Here is an example of this happening:

```
TASK: [common | reboot if required due to security updates] ******************* 
changed: [mon]
changed: [app]

TASK: [common | waiting for server to come back] ****************************** 
ok: [app -> 127.0.0.1]
ok: [mon -> 127.0.0.1]

TASK: [common | sysctl name="{{ item.name }}" value="{{ item.value }}" sysctl_set=yes state=present reload=yes] *** 
fatal: [mon] => SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh
fatal: [app] => SSH Error: data could not be sent to the remote host. Make sure this host can be reached over ssh

FATAL: all hosts have already failed -- aborting
```

**Comment by garrettr** [comment]
I think it is interesting that there are at least 3 separate tasks that implement "waiting for the server to come back". These tasks are similar, *but different*. Is it possible that these differences are the cause of the problem? I'm not sure which tasks is used when (for example, I don't know which specific task was used in the traceback @dolanjs posted in the above comment), but it seems like these differences could be involved.

Generally, we should:

1. DRY out the playbooks. I understand this is a long term project. However having 3 different (and *differing*) implementations of the same functionality is a recipe for confusion and bugs.
2. Look into whether some difference between these tasks if the a cause of the problem. For example, the significantly shorter delay on the version in `cron_apt.yml` seems suspicious to me.

```sh
➜  securedrop git:(0.3.3-release-testing-fixes) ag -A 10 "waiting for server to come back"
install_files/ansible-base/roles/common/tasks/apply_grsec_lock.yml
23:- name: waiting for server to come back
24-  local_action:
25-    module: wait_for
26-      host={{ ansible_ssh_host }}
27-      port={{ ansible_ssh_port }}
28-      delay=45
29-      state=started
30-  when: not grsec_lock.stat.exists
31-  sudo: false
32-
33-- sysctl: name="{{ item.name }}" value="{{ item.value }}" sysctl_set=yes state=present reload=yes

install_files/ansible-base/roles/common/tasks/cron_apt.yml
38:- name: waiting for server to come back
39-  local_action:
40-    module: wait_for
41-      host={{ ansible_ssh_host }}
42-      port={{ ansible_ssh_port }}
43-      delay=20
44-      state=started
45-  when: reboot_required.stat.exists
46-  sudo: false
47-


install_files/ansible-base/roles/common/tasks/from_fpf_repo_install_grsec.yml
43:- name: waiting for server to come back
44-  local_action:
45-    module: wait_for
46-      host={{ ansible_ssh_host }}
47-      port={{ ansible_ssh_port }}
48-      delay=45
49-      state=started
50-  sudo: false
51-  when: not running_grsec.stat.exists
52-  tags:
53-    - reboot
```

**Comment by conorsch** [comment]
>for example, I don't know which specific task was used in the traceback @dolanjs posted in the above comment

Actually, we do know the specific task! The line `TASK: [common | reboot if required due to security updates]` from the output shows that it's from `cron-apt.yml` in the `common` role:

```
$ grep -rl 'reboot if required due to security updates'
roles/common/tasks/cron_apt.yml
```

Your hunch that the differently figured cron-apt delay would be associated with the failure is correct, at least in this one particular instance we have the error output for. Will keep an eye out for similar occurrences, as I'm running tests on the prod VMs more and more, and provide further documentation here if the problem reoccurs.

**Comment by conorsch** [comment]
Resolved via #1700. We can now confidently reboot the servers, and even restart the tor service, when connecting over tor.
